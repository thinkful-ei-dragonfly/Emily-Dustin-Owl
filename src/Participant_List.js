import React from 'react';
import Participant from './Participant';


export default function(props) {
  return (
    <div className="participant_list">
      {props.participants.map(participant => (
      <Participant key={participant.id} name={participant.name} avatar={participant.avatar} 
      inSession={participant.inSession} onStage={participant.onStage} />))}
    </div>
    

  );
}

