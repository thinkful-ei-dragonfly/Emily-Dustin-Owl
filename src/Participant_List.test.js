import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant_List from './Participant_List';



describe('List component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    // eslint-disable-next-line react/jsx-pascal-case
    ReactDOM.render(<Participant_List participants={[
      {
        id: 1,
        name: 'Koren Templeton',
        avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
      },
      {
        id: 2,
        name: 'Caty Flucker',
        avatar:
          'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: false
      },
    ]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expexted', () => {
    const tree = renderer
      // eslint-disable-next-line react/jsx-pascal-case
      .create(<Participant_List participants={[
        {
          id: 1,
          name: 'Koren Templeton',
          avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: true
        },
        {
          id: 2,
          name: 'Caty Flucker',
          avatar:
            'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: false
        },
      ]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});