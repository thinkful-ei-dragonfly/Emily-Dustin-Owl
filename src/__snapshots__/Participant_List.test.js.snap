// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expexted 1`] = `
<div
  className="participant_list"
>
  <div
    className="Participant"
  >
    <h3>
      Koren Templeton
    </h3>
    <img
      alt="user avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <p
      className="active"
    >
      onStage
    </p>
  </div>
  <div
    className="Participant"
  >
    <h3>
      Caty Flucker
    </h3>
    <img
      alt="user avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <p
      className="active"
    >
      inSession
    </p>
  </div>
</div>
`;
